reinforcement_learning/rlclientlib/api_status.cc

```diff
@@ -28,19 +29,26 @@ namespace reinforcement_learning {
     }
   }
 
-  status_builder::status_builder(api_status* status, const int code)
-    : _code {code}, _status {status} {
-    if (status != nullptr)
+  status_builder::status_builder(i_trace* trace, api_status* status, const int code)
+    : _code { code }, _status { status }, _trace {trace} {
+    if ( enable_logging() )
       _os << "(ERR:" << _code << ")";
   }
 
   status_builder::~status_builder() {
     if (_status != nullptr) {
       api_status::try_update(_status, _code, _os.str().c_str());
     }
+    if (_trace != nullptr ) {
+      _trace->log(_os.str());
+    }
   }
 
   status_builder::operator int() const {
     return _code;
   }
+
+  bool status_builder::enable_logging() const {
+    return _status != nullptr || _trace == nullptr;
```

Comment:
>return _status != nullptr || _trace == nullptr; [](start = 4, length = 47)

Little confused by this condition: why do we enable logging if there is a api_status object or no trace object?

Reply:
Yes.  Thanks.